src/Lecture1.hs

```diff
@@ -104,7 +122,12 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Integer
+strSum str = do
+    let numList = map read (words str)
+    if null numList
+        then 0
+    else last (scanl1 (+) numList)
```

Comment:
There's a standard function `sum` that finds a sum of all elements in the list. You can use it instead. It also returns `0` on the empty list, so you don't need to have a separate check for this 🙂 

You can even write this function as a one-liner if you prefer!